Test/Invoke-WhiskeyTask.Tests.ps1

```diff
@@ -388,6 +388,42 @@ function ThenTaskRanWithoutParameter
     }
 }
 
+function ThenTempDirectoryCreated 
+{
+    param(
+        $TaskName
+    )
+
+    $expectedTempPath = Join-Path -Path $context.OutputDirectory -ChildPath ('Temp.{0}.' -f $TaskName)
+    $expectedTempPathRegex = '^{0}[a-z0-9]{{8}}\.[a-z0-9]{{3}}$' -f [regex]::escape($expectedTempPath)
+    It ('should create a task-specific temp directory') {
+        Assert-MockCalled -CommandName 'New-Item' -ModuleName 'Whiskey' -ParameterFilter { 

```

Comment:
I've always been curious why you do separate `Assert-MockCalled` for each individual parameter instead of checking all the parameters with a single `Assert-MockCalled`. Doesn't doing them separately leave it possible for this test to pass even if `New-Item` doesn't contain all the parameters in a single call?
e.g.
```
New-Item -Path $expectedTempPathRegex -ItemType 'File'
New-Item -Path 'someOtherPath' -ItemType 'Directory'
```

Reply:
Because I can't debug into the parameter filter blocks. I used to combine everything. When the assertion failed, I had to write a crap load of `Write-Debug` statements to figure out which parameter was invalid. On separate lines, I can look at the Pester failure message, see what line failed, know what parameter was wrong, and usually I can figure out why it is broken without having to add any `Write-Debug` statements. You can see here, however, that this time I couldn't. :(

Reply:
Oh that makes sense. Thanks for the explanation.